From 54d3cbaec91fbea83de36a4fc3e18054f68f05a0 Mon Sep 17 00:00:00 2001
From: Christopher Champagne <cjchampagne@habana.ai>
Date: Thu, 22 May 2025 22:30:28 +0300
Subject: [PATCH] [SW-179783] g3: refactor ethernet sensord entries

Add macros for defining the ethernet sensord entries that have copies
for each port.

Change-Id: I43b6fd393275aba52b3a87280f79091a519b088d
Signed-off-by: Christopher Champagne <cjchampagne@habana.ai>
---

diff --git a/arch/gaudi3/ras/pldm/sensord_db.c b/arch/gaudi3/ras/pldm/sensord_db.c
index 5f41ff1..9e20037 100644
--- a/arch/gaudi3/ras/pldm/sensord_db.c
+++ b/arch/gaudi3/ras/pldm/sensord_db.c
@@ -20,6 +20,57 @@
 		.api.poll = sensord_poll_ethernet_port_state,			\
 	}
 
+#define SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(port)					\
+	{											\
+		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],	\
+		.index = (port - 1),								\
+		.api.poll = sensord_poll_ethernet_correctable_errors_counter,			\
+		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,					\
+	}
+
+#define SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(port)					  \
+	{											  \
+		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER], \
+		.index = (port - 1),								  \
+		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,			  \
+	}
+
+#define SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(port)					   \
+	{											   \
+		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],\
+		.index = (port - 1),								   \
+		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,		   \
+	}
+
+#define SENSOR_ETHERNET_HIGH_BER_COUNTER(port)						\
+	{										\
+		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],	\
+		.index = (port - 1),							\
+		.api.poll = sensord_poll_ethernet_high_ber_counter,			\
+	}
+
+#define SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(port)						\
+	{											\
+		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],	\
+		.index = (port - 1),								\
+		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,				\
+	}
+
+#define SENSOR_ETHERNET_SER_POST_FEC_COUNTER(port)						\
+	{											\
+		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],	\
+		.index = (port - 1),								\
+		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,				\
+	}
+
+#define SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(port)						\
+	{											\
+		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],	\
+		.index = (port - 1),								\
+		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,			\
+		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,					\
+	}
+
 struct sensord_sensor_entry sensord_db[] = {
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_CURRENT_ADC_48V_POWER],
@@ -404,1063 +455,174 @@
 		.api.poll = sensord_poll_ethernet_serdes_link_speed,
 		.api.poll_once = true,
 	},
-
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 0,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 1,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 2,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 3,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 4,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 5,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 6,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 7,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 8,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 9,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 10,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 11,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 12,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 13,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 14,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 15,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 16,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 17,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 18,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 19,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 20,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 21,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 22,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_correctable_errors_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 23,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 0,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 1,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 2,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 3,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 4,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 5,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 6,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 7,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 8,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 9,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 10,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 11,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 12,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 13,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 14,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 15,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 16,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 17,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 18,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 19,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 20,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 21,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 22,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER],
-		.api.poll = sensord_poll_ethernet_uncorrectable_errors_counter,
-		.index = 23,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 0,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 1,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 2,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 3,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 4,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 5,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 6,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 7,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 8,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 9,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 10,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 11,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 12,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 13,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 14,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 15,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 16,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 17,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 18,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 19,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 20,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 21,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 22,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER],
-		.api.poll = sensord_poll_ethernet_link_re_initialization_counter,
-		.index = 23,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 0,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 1,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 2,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 3,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 4,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 5,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 6,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 7,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 8,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 9,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 10,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 11,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 12,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 13,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 14,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 15,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 16,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 17,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 18,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 19,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 20,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 21,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 22,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_HIGH_BER_COUNTER],
-		.api.poll = sensord_poll_ethernet_high_ber_counter,
-		.index = 23,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 0,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 1,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 2,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 3,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 4,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 5,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 6,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 7,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 8,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 9,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 10,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 11,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 12,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 13,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 14,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 15,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 16,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 17,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 18,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 19,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 20,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 21,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 22,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_PRE_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_pre_fec_counter,
-		.index = 23,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 0,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 1,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 2,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 3,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 4,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 5,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 6,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 7,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 8,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 9,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 10,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 11,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 12,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 13,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 14,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 15,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 16,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 17,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 18,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 19,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 20,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 21,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 22,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_SER_POST_FEC_COUNTER],
-		.api.poll = sensord_poll_ethernet_ser_post_fec_counter,
-		.index = 23,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 0,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 1,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 2,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 3,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 4,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 5,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 6,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 7,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 8,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 9,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 10,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 11,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 12,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 13,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 14,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 15,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 16,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 17,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 18,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 19,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 20,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 21,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 22,
-	},
-	{
-		.pdr = &pldm_pdr_records[
-			PLDM_RIDX_SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER],
-		.api.poll = sensord_poll_ethernet_ports_link_toggle_counter,
-		.api.polling_rate_ms = SENSOR_SAMPLE_RATE_HIGH,
-		.index = 23,
-	},
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(1),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(2),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(3),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(4),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(5),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(6),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(7),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(8),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(9),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(10),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(11),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(12),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(13),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(14),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(15),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(16),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(17),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(18),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(19),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(20),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(21),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(22),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(23),
+	SENSOR_ETHERNET_CORRECTABLE_ERRORS_COUNTER(24),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(1),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(2),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(3),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(4),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(5),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(6),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(7),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(8),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(9),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(10),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(11),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(12),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(13),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(14),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(15),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(16),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(17),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(18),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(19),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(20),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(21),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(22),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(23),
+	SENSOR_ETHERNET_UNCORRECTABLE_ERRORS_COUNTER(24),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(1),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(2),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(3),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(4),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(5),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(6),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(7),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(8),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(9),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(10),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(11),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(12),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(13),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(14),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(15),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(16),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(17),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(18),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(19),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(20),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(21),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(22),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(23),
+	SENSOR_ETHERNET_LINK_RE_INITIALIZATION_COUNTER(24),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(1),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(2),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(3),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(4),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(5),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(6),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(7),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(8),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(9),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(10),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(11),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(12),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(13),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(14),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(15),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(16),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(17),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(18),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(19),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(20),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(21),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(22),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(23),
+	SENSOR_ETHERNET_HIGH_BER_COUNTER(24),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(1),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(2),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(3),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(4),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(5),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(6),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(7),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(8),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(9),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(10),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(11),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(12),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(13),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(14),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(15),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(16),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(17),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(18),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(19),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(20),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(21),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(22),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(23),
+	SENSOR_ETHERNET_SER_PRE_FEC_COUNTER(24),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(1),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(2),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(3),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(4),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(5),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(6),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(7),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(8),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(9),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(10),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(11),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(12),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(13),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(14),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(15),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(16),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(17),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(18),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(19),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(20),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(21),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(22),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(23),
+	SENSOR_ETHERNET_SER_POST_FEC_COUNTER(24),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(1),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(2),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(3),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(4),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(5),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(6),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(7),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(8),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(9),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(10),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(11),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(12),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(13),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(14),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(15),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(16),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(17),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(18),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(19),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(20),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(21),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(22),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(23),
+	SENSOR_ETHERNET_PORTS_LINK_TOGGLE_COUNTER(24),
 	{
 		.pdr = &pldm_pdr_records[PLDM_RIDX_SENSOR_ETHERNET_CONFIG],
 		.api.poll = sensord_poll_ethernet_config,
